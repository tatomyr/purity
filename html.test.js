const { html } = require('./.__html__.js')

const Parent = ({ children }) => `
  <div style="background-color: red; padding: 10px;">
    ${children || ''}
  </div>
`

const Child = ({ text, color, price }) => `
  <h2 style="background-color: ${color}; text-transform: capitalize;">
    ${text} | $${price}
  </h2>
`

describe('html', () => {
  it('should match a snapshot when Parent is empty', () => {
    const Component = () => html(Parent)`<Parent />`
    expect(Component()).toMatchSnapshot()
  })
  it('should match a snapshot when Parent containts a regular html tag', () => {
    const Component = () => html(Parent)`
      <Parent children="<h1 style='background-color: green;'>Inner text</h1>" />
    `
    expect(Component()).toMatchSnapshot()
  })
  it('should match a snapshot when Parent contains a Child component', () => {
    const Component = () => html(Parent, Child)`
      <Parent
        children="
          <Child
            price=${1.1 + 2.2}
            color='yellow'
            text='Hello!'
          />
        "
      />
    `
    expect(Component()).toMatchSnapshot()
  })
  it('should match a snapshot when Parent contains a tag and a component', () => {
    const Component = () => html(Parent, Child)`
      <Parent
        children="
          <Child
            price=${1.1 + 2.2}
            color='yellow'
            text='Hello!'
          />
          <h1 style='background-color: green;'>Inner text</h1>
        "
      />
    `
    expect(Component()).toMatchSnapshot()
  })
})
