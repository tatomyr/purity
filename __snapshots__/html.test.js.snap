// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html should match a snapshot when Parent contains a Child component 1`] = `
"
      
  <div style=\\"background-color: red; padding: 10px;\\">
                <h2 style=\\"background-color: yellow; text-transform: capitalize;\\">    Hello! | $3.3000000000000003  </h2>        
  </div>

    "
`;

exports[`html should match a snapshot when Parent contains a tag and a component 1`] = `
"
      
  <div style=\\"background-color: red; padding: 10px;\\">
                <h2 style=\\"background-color: yellow; text-transform: capitalize;\\">    Hello! | $3.3000000000000003  </h2>          <h1 style='background-color: green;'>Inner text</h1>        
  </div>

    "
`;

exports[`html should match a snapshot when Parent containts a regular html tag 1`] = `
"
      
  <div style=\\"background-color: red; padding: 10px;\\">
    <h1 style='background-color: green;'>Inner text</h1>
  </div>

    "
`;

exports[`html should match a snapshot when Parent is empty 1`] = `
"
  <div style=\\"background-color: red; padding: 10px;\\">
    
  </div>
"
`;
